src/App.tsx

```diff
@@ -15,36 +15,110 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+const FIELD_ALPHABETIC = 'alphabetically';
+const FIELD_LENGTH = 'length';
+
 export const App: React.FC = () => {
+  const [goods, setGoods] = useState<string[]>(goodsFromServer);
+  const [sortBy, setSortBy] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+
+  const getSortedGoods = (orderBy: string, reverse = false) => {
+    const preparedGoods = [...goodsFromServer];
+
+    switch (orderBy) {
+      case FIELD_ALPHABETIC:
+        preparedGoods.sort((a, b) => a.localeCompare(b));
+        break;
+      case FIELD_LENGTH:
+        preparedGoods.sort((a, b) => a.length - b.length);
+        break;
+      default:
+        break;
+    }
+
+    if (reverse) {
+      preparedGoods.reverse();
+    }
+
+    return preparedGoods;
+  };
+
+  const handleSortAlphabetically = () => {
+    const sorted = getSortedGoods(FIELD_ALPHABETIC, isReversed);
+
+    setGoods(sorted);
+    setSortBy(FIELD_ALPHABETIC);
+  };
+
+  const handleSortByLength = () => {
+    const sorted = getSortedGoods(FIELD_LENGTH, isReversed);
+
+    setGoods(sorted);
+    setSortBy(FIELD_LENGTH);
+  };
+
+  const handleReverse = () => {
+    const reversed = [...goods].reverse();
+
+    setGoods(reversed);
+    setIsReversed(prev => !prev);
+  };
```

Comment:
handlers shouldn't do such work
they should just change filter state
all work is doing in getSortedGoods outside handlers

```suggestion
  const handleSortByLength = () => {
    setSortBy(FIELD_LENGTH);
  };

  const handleReverse = () => {
    setGoods(reversed);
  };
```